File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/experiment.py
# coding: utf-8
# Copyright (c) 2016, 2026, Oracle and/or its affiliates.  All rights reserved.
# This software is dual-licensed to you under the Universal Permissive License (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License 2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose either license.


from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class Experiment(object):
    """
    Details of an MLflow experiment.
    """

    def __init__(self, **kwargs):
        """
        Initializes a new Experiment object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param experiment_id:
            The value to assign to the experiment_id property of this Experiment.
        :type experiment_id: str

        :param name:
            The value to assign to the name property of this Experiment.
        :type name: str

        :param artifact_location:
            The value to assign to the artifact_location property of this Experiment.
        :type artifact_location: str

        :param lifecycle_stage:
            The value to assign to the lifecycle_stage property of this Experiment.
        :type lifecycle_stage: str

        :param creation_time:
            The value to assign to the creation_time property of this Experiment.
        :type creation_time: int

        :param last_update_time:
            The value to assign to the last_update_time property of this Experiment.
        :type last_update_time: int

        :param tags:
            The value to assign to the tags property of this Experiment.
        :type tags: list[oci.aidataplatform_dp.models.ExperimentTag]

        """
        self.swagger_types = {
            'experiment_id': 'str',
            'name': 'str',
            'artifact_location': 'str',
            'lifecycle_stage': 'str',
            'creation_time': 'int',
            'last_update_time': 'int',
            'tags': 'list[ExperimentTag]'
        }

        self.attribute_map = {
            'experiment_id': 'experiment_id',
            'name': 'name',
            'artifact_location': 'artifact_location',
            'lifecycle_stage': 'lifecycle_stage',
            'creation_time': 'creation_time',
            'last_update_time': 'last_update_time',
            'tags': 'tags'
        }

        self._experiment_id = None
        self._name = None
        self._artifact_location = None
        self._lifecycle_stage = None
        self._creation_time = None
        self._last_update_time = None
        self._tags = None

    @property
    def experiment_id(self):
        """
        **[Required]** Gets the experiment_id of this Experiment.
        Unique identifier for the experiment.


        :return: The experiment_id of this Experiment.
        :rtype: str
        """
        return self._experiment_id

    @experiment_id.setter
    def experiment_id(self, experiment_id):
        """
        Sets the experiment_id of this Experiment.
        Unique identifier for the experiment.


        :param experiment_id: The experiment_id of this Experiment.
        :type: str
        """
        self._experiment_id = experiment_id

    @property
    def name(self):
        """
        **[Required]** Gets the name of this Experiment.
        Name of the experiment.


        :return: The name of this Experiment.
        :rtype: str
        """
        return self._name

    @name.setter
    def name(self, name):
        """
        Sets the name of this Experiment.
        Name of the experiment.


        :param name: The name of this Experiment.
        :type: str
        """
        self._name = name

    @property
    def artifact_location(self):
        """
        Gets the artifact_location of this Experiment.
        Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.


        :return: The artifact_location of this Experiment.
        :rtype: str
        """
        return self._artifact_location

    @artifact_location.setter
    def artifact_location(self, artifact_location):
        """
        Sets the artifact_location of this Experiment.
        Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.


        :param artifact_location: The artifact_location of this Experiment.
        :type: str
        """
        self._artifact_location = artifact_location

    @property
    def lifecycle_stage(self):
        """
        Gets the lifecycle_stage of this Experiment.
        Lifecycle stage of the experiment, e.g., 'active' or 'deleted'.


        :return: The lifecycle_stage of this Experiment.
        :rtype: str
        """
        return self._lifecycle_stage

    @lifecycle_stage.setter
    def lifecycle_stage(self, lifecycle_stage):
        """
        Sets the lifecycle_stage of this Experiment.
        Lifecycle stage of the experiment, e.g., 'active' or 'deleted'.


        :param lifecycle_stage: The lifecycle_stage of this Experiment.
        :type: str
        """
        self._lifecycle_stage = lifecycle_stage

    @property
    def creation_time(self):
        """
        Gets the creation_time of this Experiment.
        Unix timestamp in milliseconds when the experiment was created.


        :return: The creation_time of this Experiment.
        :rtype: int
        """
        return self._creation_time

    @creation_time.setter
    def creation_time(self, creation_time):
        """
        Sets the creation_time of this Experiment.
        Unix timestamp in milliseconds when the experiment was created.


        :param creation_time: The creation_time of this Experiment.
        :type: int
        """
        self._creation_time = creation_time

    @property
    def last_update_time(self):
        """
        Gets the last_update_time of this Experiment.
        Unix timestamp in milliseconds when the experiment was last updated.


        :return: The last_update_time of this Experiment.
        :rtype: int
        """
        return self._last_update_time

    @last_update_time.setter
    def last_update_time(self, last_update_time):
        """
        Sets the last_update_time of this Experiment.
        Unix timestamp in milliseconds when the experiment was last updated.


        :param last_update_time: The last_update_time of this Experiment.
        :type: int
        """
        self._last_update_time = last_update_time

    @property
    def tags(self):
        """
        Gets the tags of this Experiment.
        List of tags set on the experiment.


        :return: The tags of this Experiment.
        :rtype: list[oci.aidataplatform_dp.models.ExperimentTag]
        """
        return self._tags

    @tags.setter
    def tags(self, tags):
        """
        Sets the tags of this Experiment.
        List of tags set on the experiment.


        :param tags: The tags of this Experiment.
        :type: list[oci.aidataplatform_dp.models.ExperimentTag]
        """
        self._tags = tags

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client-2x/src/main/java/com/oracle/aidataplatform/dp/model/Experiment.java
package com.oracle.aidataplatform.dp.model;


/**
 * Details of an MLflow experiment.
**/
@javax.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=Experiment.Builder.class)

public final class Experiment  {
    @Deprecated
    @java.beans.ConstructorProperties({"experimentId", "name", "artifactLocation", "lifecycleStage", "creationTime", "lastUpdateTime", "tags"})
    public Experiment(String experimentId, String name, String artifactLocation, String lifecycleStage, Long creationTime, Long lastUpdateTime, java.util.List<ExperimentTag> tags) {
        super();
        this.experimentId = experimentId;
        this.name = name;
        this.artifactLocation = artifactLocation;
        this.lifecycleStage = lifecycleStage;
        this.creationTime = creationTime;
        this.lastUpdateTime = lastUpdateTime;
        this.tags = tags;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
            /**
     * Unique identifier for the experiment.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("experiment_id")
        private String experimentId;

                /**
         * Unique identifier for the experiment.
         * @param experimentId the value to set
         * @return this builder
         **/
        

        public Builder experimentId(String experimentId) {
        this.experimentId = experimentId;
        return this;
        }
            /**
     * Name of the experiment.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("name")
        private String name;

                /**
         * Name of the experiment.
         * @param name the value to set
         * @return this builder
         **/
        

        public Builder name(String name) {
        this.name = name;
        return this;
        }
            /**
     * Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("artifact_location")
        private String artifactLocation;

                /**
         * Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.
         * @param artifactLocation the value to set
         * @return this builder
         **/
        

        public Builder artifactLocation(String artifactLocation) {
        this.artifactLocation = artifactLocation;
        return this;
        }
            /**
     * Lifecycle stage of the experiment, e.g., 'active' or 'deleted'.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("lifecycle_stage")
        private String lifecycleStage;

                /**
         * Lifecycle stage of the experiment, e.g., 'active' or 'deleted'.
         * @param lifecycleStage the value to set
         * @return this builder
         **/
        

        public Builder lifecycleStage(String lifecycleStage) {
        this.lifecycleStage = lifecycleStage;
        return this;
        }
            /**
     * Unix timestamp in milliseconds when the experiment was created.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("creation_time")
        private Long creationTime;

                /**
         * Unix timestamp in milliseconds when the experiment was created.
         * @param creationTime the value to set
         * @return this builder
         **/
        

        public Builder creationTime(Long creationTime) {
        this.creationTime = creationTime;
        return this;
        }
            /**
     * Unix timestamp in milliseconds when the experiment was last updated.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("last_update_time")
        private Long lastUpdateTime;

                /**
         * Unix timestamp in milliseconds when the experiment was last updated.
         * @param lastUpdateTime the value to set
         * @return this builder
         **/
        

        public Builder lastUpdateTime(Long lastUpdateTime) {
        this.lastUpdateTime = lastUpdateTime;
        return this;
        }
            /**
     * List of tags set on the experiment.
     **/
    
        @com.fasterxml.jackson.annotation.JsonProperty("tags")
        private java.util.List<ExperimentTag> tags;

                /**
         * List of tags set on the experiment.
         * @param tags the value to set
         * @return this builder
         **/
        

        public Builder tags(java.util.List<ExperimentTag> tags) {
        this.tags = tags;
        return this;
        }


        public Experiment build() {
            Experiment model = new Experiment(this.experimentId
                    , this.name
                    , this.artifactLocation
                    , this.lifecycleStage
                    , this.creationTime
                    , this.lastUpdateTime
                    , this.tags);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(Experiment model) {
                this.experimentId(model.getExperimentId());
    this.name(model.getName());
    this.artifactLocation(model.getArtifactLocation());
    this.lifecycleStage(model.getLifecycleStage());
    this.creationTime(model.getCreationTime());
    this.lastUpdateTime(model.getLastUpdateTime());
    this.tags(model.getTags());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Unique identifier for the experiment.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("experiment_id")
    private final String experimentId;

        /**
     * Unique identifier for the experiment.
     * @return the value
     **/
    
    public String getExperimentId() {
        return experimentId;
    }


        /**
     * Name of the experiment.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("name")
    private final String name;

        /**
     * Name of the experiment.
     * @return the value
     **/
    
    public String getName() {
        return name;
    }


        /**
     * Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("artifact_location")
    private final String artifactLocation;

        /**
     * Location where all artifacts for the experiment are stored. If not provided, the remote server will select an appropriate default.
     * @return the value
     **/
    
    public String getArtifactLocation() {
        return artifactLocation;
    }


        /**
     * Lifecycle stage of the experiment, e.g., 'active' or 'deleted'.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("lifecycle_stage")
    private final String lifecycleStage;

        /**
     * Lifecycle stage of the experiment, e.g., 'active' or 'deleted'.
     * @return the value
     **/
    
    public String getLifecycleStage() {
        return lifecycleStage;
    }


        /**
     * Unix timestamp in milliseconds when the experiment was created.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("creation_time")
    private final Long creationTime;

        /**
     * Unix timestamp in milliseconds when the experiment was created.
     * @return the value
     **/
    
    public Long getCreationTime() {
        return creationTime;
    }


        /**
     * Unix timestamp in milliseconds when the experiment was last updated.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("last_update_time")
    private final Long lastUpdateTime;

        /**
     * Unix timestamp in milliseconds when the experiment was last updated.
     * @return the value
     **/
    
    public Long getLastUpdateTime() {
        return lastUpdateTime;
    }


        /**
     * List of tags set on the experiment.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("tags")
    private final java.util.List<ExperimentTag> tags;

        /**
     * List of tags set on the experiment.
     * @return the value
     **/
    
    public java.util.List<ExperimentTag> getTags() {
        return tags;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("Experiment(");
        sb.append("experimentId=").append(String.valueOf(this.experimentId));
        sb.append(", name=").append(String.valueOf(this.name));
        sb.append(", artifactLocation=").append(String.valueOf(this.artifactLocation));
        sb.append(", lifecycleStage=").append(String.valueOf(this.lifecycleStage));
        sb.append(", creationTime=").append(String.valueOf(this.creationTime));
        sb.append(", lastUpdateTime=").append(String.valueOf(this.lastUpdateTime));
        sb.append(", tags=").append(String.valueOf(this.tags));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof Experiment)) {
            return false;
        }

        Experiment other = (Experiment) o;
        return java.util.Objects.equals(this.experimentId, other.experimentId) &&
            java.util.Objects.equals(this.name, other.name) &&
            java.util.Objects.equals(this.artifactLocation, other.artifactLocation) &&
            java.util.Objects.equals(this.lifecycleStage, other.lifecycleStage) &&
            java.util.Objects.equals(this.creationTime, other.creationTime) &&
            java.util.Objects.equals(this.lastUpdateTime, other.lastUpdateTime) &&
            java.util.Objects.equals(this.tags, other.tags);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.experimentId == null ? 43 : this.experimentId.hashCode());
        result = (result * PRIME) + (this.name == null ? 43 : this.name.hashCode());
        result = (result * PRIME) + (this.artifactLocation == null ? 43 : this.artifactLocation.hashCode());
        result = (result * PRIME) + (this.lifecycleStage == null ? 43 : this.lifecycleStage.hashCode());
        result = (result * PRIME) + (this.creationTime == null ? 43 : this.creationTime.hashCode());
        result = (result * PRIME) + (this.lastUpdateTime == null ? 43 : this.lastUpdateTime.hashCode());
        result = (result * PRIME) + (this.tags == null ? 43 : this.tags.hashCode());
        return result;
    }


}
